modules/apps/headless/headless-delivery/headless-delivery-test/src/testFunctional/macros/DocumentFolderAPI.macro

```diff
@@ -76,4 +165,48 @@ definition {
 		return ${response};
 	}
 
+	macro getDocumentFoldersWithDifferentPameter {
+		var curl = DocumentFolderAPI._curlDocumentFolders(
+			groupName = ${groupName},
+			parameter = ${parameter},
+			parameterValue = ${parameterValue});
+
+		var response = JSONCurlUtil.get(${curl});
+
+		return ${response};
+	}
+
+	macro setUpGlobalDocumentFolder {
+		Variables.assertDefined(parameterList = ${groupName});
```

Comment:
As shared, maybe we can make setUpGlobalDocumentFolderIdAndErc by passing response as variable if any needs in the future :smile:

Reply:
I think if would divide here the creation of the folder and making the parameters static. The reason is that if on the testcase level I see a method setUpGlobalDocumentFolder I don't really know what variables are being created and used. 
So on the testcase level I'd put first:

```
var response = DocumentFolderAPI.addDocumentFolder(
			externalReferenceCode = ${externalReferenceCode},
			groupName = ${groupName},
			name = ${name});

```
And then I would put two macros: `staticDoucmentFolderId(response=${response}) `and `staticDoucmentFolderErc(response=${response})` 
This way it looks more transaprent in the testcase and we will know where the staticDoucmentFolderId is appearing from in the test. It's pretty hard for debugging if it is not visible.

Reply:
@magjed4289 I think it's a better way, and how about setting the method name to "setUpGlobalDocumentFolderId" and "setUpGlobalDocumentFolderErc"?

Reply:
I would just leave staticDoucmentFolderId and staticDoucmentFolderErc as these are the names of the variables that we are returning with these macros. It is easier, later on, to know where are they appearing from in the test case code. Does that make sense?